tests/python_package_test/test_utilities.py

```diff
@@ -29,17 +29,18 @@ def dummy_metric(_, __):
                   [1, 2, 3]],
                  dtype=np.float32)
     y = np.array([0, 1, 1, 0])
-    lgb_data = lgb.Dataset(X, y)
+    lgb_train = lgb.Dataset(X, y)
+    lgb_valid = lgb.Dataset(X, y)  # different object for early-stopping
 
     eval_records = {}
     callbacks = [
         lgb.record_evaluation(eval_records),
         lgb.log_evaluation(2),
-        lgb.early_stopping(4)
+        lgb.early_stopping(10)
```

Comment:
I changed this because otherwise it would stop at the 4th iteration, this was due to the training set being ignored for early stopping but now early stopping is disabled if you only provide the training set as valid, so I had to provide a dummy validation set.